architecture/Metatransactions.md

```diff
@@ -0,0 +1,68 @@
+# Metatransactions in the CDapp
+
+Metatransactions (MetaTX) present a paradigm shift in the blockchain world, especially in enhancing user experience by removing the need for users to pay gas fees. The CDapp leverages this concept by introducing a two-pronged approach involving the broadcaster service and specially designed contract methods.
+
+## What are Metatransactions?
+
+Metatransactions encapsulate the original transaction and relay it via a third-party entity which takes responsibility for the gas fee. This enables seamless interactions, particularly beneficial for new users unfamiliar with the concept of gas or those unwilling to spend on every transaction.
+
+### Benefits
+
+User Friendliness: Users without ETH can still perform transactions.
+Flexibility: Allows dApps like CDapp to decide when and how to bear the gas costs, possibly leading to promotional or free transaction periods.
+Adoption: Removes a major hurdle for new users by simplifying interactions with blockchain-based applications.
+
+## Broadcaster Service Middleware
+
+The Broadcaster Service is not just an intermediary, but a critical component in the Metatransaction architecture.
+
+### Workflow
+- The user initiates a transaction on the CDapp.
+- Instead of sending the transaction directly to the blockchain, it's encoded and packaged.
+- This encoded package is then sent to the Broadcaster Service.
+- The Broadcaster Service deciphers the package, recreating the original transaction.
+- The service then transmits the transaction to the blockchain, paying the required gas fees.
+
+```
+ CDapp
+   |
+   | (Encoded Transaction Data)
+   v
+Broadcaster Service (Decodes & Broadcasts Transaction)
+   |
+   | (Reconstructed Transaction)
+   v
+Blockchain
+```
+
+### Security & Trust
+
+By design, the broadcaster does not have access to the user's private key. Instead, it merely acts on the user's behalf, using its funds to process the transaction. This ensures that while the transaction integrity is maintained, the user's security is not compromised.
+
+### Maintenance
+
+The Broadcaster Service's wallet requires periodic top-ups. It's crucial to monitor the balance to prevent service disruptions. Automated alerts are set up to notify us in the case of the wallet balance getting low.
+
+## The Contracts
+
+To facilitate metatransactions, the Colony Network introduced specialized methods after v9 in their contracts.
```

Comment:
What are these methods? Can you give more detail into them and/or a link to the code?

Reply:
Or if the specialised methods you are referring to here are the `Authorization Mechanism` section below, it would be good to make that more explicit